op-node/p2p/discovery.go

```diff
@@ -417,6 +426,17 @@ func (n *NodeP2P) DiscoveryProcess(ctx context.Context, log log.Logger, cfg *rol
 	}
 }
 
+// recordPeerMD saves extended metadata about a peer to the peerstore.
+// if any of the metadata fails to be saved, it is logged and ignored.
+func recordPeerMD(pstore peerstore.Peerstore, id peer.ID, md map[string]interface{}) {
+	for k, v := range md {
+		gob.Register(v)
+		if err := pstore.Put(id, k, v); err != nil {
+			log.Warn("failed to record peer metadata", "key", k, "value", v, "err", err)
+		}
+	}
+}
```

Comment:
Great point, every peer added will make two calls to gob.Register, and after the first peer, each call will be redundant.

I had done it this way to hide the complexity of loading arbitrary types into the store. However, since we are presently only using `string` and `uint64`, I will simply remove this `gob.Register` call and leave a comment in its place if/when it needs to be extended.